dipy/segment/featurespeed.pxd

```diff
@@ -0,0 +1,19 @@
+from cythonutils cimport Data2D, Shape
+
+
+cdef class Feature(object):
+    cdef int is_order_invariant
+
+    cdef Shape c_infer_shape(Feature self, Data2D streamline) nogil except *
```

Comment:
Question. Although you have defined DataXD here you assume Data2D. Is there a reason for that?
Also, streamline is only specific for streamlines should you use a more generic variable name here?